docs/sql-ref-syntax-qry-select-groupby.md

```diff
@@ -93,6 +95,38 @@ aggregate_name ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE boolean_ex
      (product, warehouse, location), (warehouse), (product), (warehouse, product), ())`.
     The N elements of a `CUBE` specification results in 2^N `GROUPING SETS`.
 
+* **Partial Grouping Analytics**
+
+    Partial grouping analytics means there are both `group_expression` and `CUBE|ROLLUP|GROUPING SETS`
+    in GROUP BY clause. `CUBE` and `ROLLUP` are just syntax sugar for GROUPING SETS, for how to use
+    `CUBE` and `ROLLUP` can refer to the section above about `CUBE` and `ROLLUP`. For example:
+    `GROUP BY warehouse, GROUPING SETS((product, location), (product), (location), ())` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, product), (warehouse, location), (warehouse))`.
+    `GROUP BY warehouse, GROUPING SETS((product, location), (producet), ())` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, location), (warehouse))`.
+
+* **Concatenated Grouping Analytics**
```

Comment:
I think we can combine these 2 new sections:
```
* **Mixed Grouping Analytics**
A GROUP BY clause can include multiple  `group_expression`s and multiple `CUBE|ROLLUP|GROUPING SETS`s.
`CUBE|ROLLUP` is just a syntax sugar for `GROUPING SETS`, please refer to the sections above for
how to translate `CUBE|ROLLUP` to `GROUPING SETS`. `group_expression` can be treated as a single-group
`GROUPING SETS` under this context. For multiple `GROUPING SETS` in the `GROUP BY` clause, we generate
a single `GROUPING SETS` by doing a cross-product of the original `GROUPING SETS`s. For example, ...
```

Reply:
Both `CUBE|ROLLUP` and `group_expression` can be translated to GROUPING SETS under this context, then it's much easier to describe the semantic.

Reply:
Done